Simplenote/src/main/java/com/automattic/simplenote/utils/AuthUtils.java

```diff
@@ -55,4 +63,30 @@ public static void logOut(Simplenote application) {
 
         WidgetUtils.updateNoteWidgets(application);
     }
+
+    public static boolean hasDifferentEmail(Simplenote application, Uri uri) {
+        User user = application.getSimperium().getUser();
+        return user.getStatus().equals(User.Status.AUTHORIZED) &&
+            !user.getEmail().equals(extractEmailFromMagicLink(uri));
+    }
+
+    public static void magicLinkLogin(Simplenote application, Uri uri) {
+        String userEmail = extractEmailFromMagicLink(uri);
+        String spToken = uri.getQueryParameter("token");
+
+        User user = application.getSimperium().getUser();
+        user.setAccessToken(spToken);
+        user.setEmail(userEmail);
+        user.setStatus(User.Status.AUTHORIZED);
```

Comment:
on a similar note would it make sense to pass control back to the main application? I know all this was previously handled by the Simperium library itself so it all surely is a bit awkward.

```java
(Simplenote) getApplication.login(userEmail, spToken);
```

Reply:
This is now done here: d4c12c8dc9748f950ef68d51ddeaf59c247ad8e1